src/Perpetuum/Services/EventServices/EventProcessors/NpcStateAnnouncer.cs

```diff
@@ -51,7 +52,16 @@ private int GetNpcDef(NpcStateMessage msg)
         private string GetNpcName(int def)
         {
             var nameToken = EntityDefault.Get(def).Name + "_name";
-            var name = _nameDictionary[nameToken]?.ToString();
+            var name = "";
+            try
+            {
+                name = _nameDictionary[nameToken]?.ToString();
+            }
+            catch (KeyNotFoundException)
+            {
+                name = "NameNotFound";
+            }
```

Comment:
2 things to think about here:
First:
There might yet be a better way: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=net-6.0

I am usually of the mind that (unless avoidable by construct of the api/interface of the call) to avoid throwing the exception and handle it in non-exceptional controlflow either with a `TryX() else{..error handle..}` or `GetOrDefault()`-> and then handle the empty/null return case.

Second:
And I do not know if the best solution is to print "NameNotFound" in the channel for players to see.  If this happens in production it will make no sense to them and not help them find bosses spawning as intended.
Let's instead enhance the log-viewer's experience to correct this error by pointing them in the right direction to fix the configuration: 
For example: "Missing localized name for :{nameToken}" would be super informative for an admin to catch this in the logs, and add this missing value to the customDictionary.

Your thoughts?

Reply:
TryGetValue does seem like a better solution and I agree that the message in the channel could be better. What you've suggested is better than what I can come up with.